e2echart/e2e-chart-template.html

```diff
@@ -97,13 +97,50 @@ <h5 class="modal-title">Resource</h5>
         return false
     }
 
+    // If this is a leader found, lost, elected, or missing, return true and the reason.
+    // Otherwise, return false.
+    function isEtcdLeaderFoundOrLost(eventInterval) {
+        const { tempStructuredLocator, tempStructuredMessage } = eventInterval;
+        let reason = null;
+        if (
+            tempStructuredLocator &&
+            tempStructuredLocator.keys &&
+            tempStructuredLocator.keys.hasOwnProperty('namespace') &&
+            tempStructuredLocator.keys['namespace'].includes('openshift-etcd')
+        ) {
+            if (tempStructuredMessage && tempStructuredMessage.hasOwnProperty('reason')) {
+                reason = tempStructuredMessage['reason']
+                if (reason.includes('leader-found') || reason.includes('leader-lost') ||
+                    reason.includes('leader-elected') || reason.includes('leader-missing')) {
```

Comment:
Could we instead update where these intervals are created (after #28474 merges) to have tempSource = EtcdLeadership, and eliminate the need for reason checking here in the js.